packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -705,11 +708,13 @@ export class ComboBox extends React.Component<IComboBoxProps, IComboBoxState> {
     const displayValues = [];
     for (let idx = 0; selectedIndices && idx < selectedIndices.length; idx++) {
       const index: number = selectedIndices[idx];
-      displayValues.push(
-        this._indexWithinBounds(currentOptions, index)
-          ? currentOptions[index].text
-          : this._normalizeToString(suggestedDisplayValue),
-      );
+      if (currentOptions[index].itemType !== SelectableOptionMenuItemType.SelectAll) {
```

Comment:
Nit: When taking action based on enums, it can be safer to have a switch statement.  This indicates which values you knew about at the time and you can report a warning or error if you don't match an expected value.